docs/book/src/roadmap.md

```diff
@@ -0,0 +1,42 @@
+# Cluster API Roadmap
+
+This roadmap is a constant work in progress, subject to frequent revision. Dates are approximations.
+
+## Ongoing
+
+- Documentation improvements
+
+## v0.3 (v1alpha3) ~ March 2020
+
+- Control plane provider concept, KubeadmControlPlane implementation
+- clusterctl v2 - easier installation & upgrades of Cluster API + providers
+- Integration with cloud provider VM scaling groups (MachinePools)
+- Machine failure remediation
+- Provider end to end testing framework
+
+## v0.4 (v1alpha4) ~ July 2020
+
+- [Machine bootstrap failure detection](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/972)
+- [Improved status conditions](https://github.com/kubernetes-sigs/cluster-api/issues/1658)
+- [Machine load balancers](https://github.com/kubernetes-sigs/cluster-api/issues/1250)
+- [Extensible machine pre-delete hooks](https://github.com/kubernetes-sigs/cluster-api/issues/1514)
+
+## v0.5 (v1???) ~ November 2020
+
+- ?
+
+## TBD
+
+- [Pluggable MachineDeployment upgrade strategies](https://github.com/kubernetes-sigs/cluster-api/issues/1754)
+- Dual stack IPv4/IPv6 support
+- Windows nodes
+- Auto-scaling support (either with cluster-autoscaler, or something else)
+- UEFI booting
+- GPU support (BitFusion, Elastic GPU, etc.)
```

Comment:
This also seems like something that would currently be more machine infrastructure provider specific.

Reply:
Maybe? Probably? 😄 I don't know if there's anything that would make sense for a generic/specific split here for GPUs, or if it's 100% on the provider types & code.

Reply:
Above is CAPV and CAPA in particular.

Reply:
For image-builder, most often the drivers for this stuff are proprietary - so would need to make this stuff modular/as addons, and check licenses with regards to providing the ansible scripts for installation.